modules/stochastic_tools/src/surrogates/BasicNNTrainer.C

```diff
@@ -0,0 +1,185 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BasicNNTrainer.h"
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsApp", BasicNNTrainer);
+
+InputParameters
+BasicNNTrainer::validParams()
+{
+  InputParameters params = SurrogateTrainer::validParams();
+
+  params.addClassDescription("Trains a simple neural network using libtorch.");
+
+  MooseEnum data_type("real=0 vector_real=1", "real");
+  params.addRequiredParam<ReporterName>(
+      "response",
+      "Reporter value of response results, can be vpp with <vpp_name>/<vector_name> or sampler "
+      "column with 'sampler/col_<index>'.");
+  params.addParam<MooseEnum>("response_type", data_type, "Response data type.");
+  params.addParam<std::vector<ReporterName>>(
+      "predictors",
+      std::vector<ReporterName>(),
+      "Reporter values used as the independent random variables, If 'predictors' and "
+      "'predictor_cols' are both empty, all sampler columns are used.");
+  params.addParam<std::vector<unsigned int>>(
+      "predictor_cols",
+      std::vector<unsigned int>(),
+      "Sampler columns used as the independent random variables, If 'predictors' and "
+      "'predictor_cols' are both empty, all sampler columns are used.");
+  params.addParam<unsigned int>("no_batches", 1, "Number of batches.");
+  params.addParam<unsigned int>("no_epochs", 1, "Number of epochs.");
+  params.addParam<unsigned int>("no_hidden_layers", 0, "Number of hidden layers.");
+  params.addParam<std::vector<unsigned int>>(
+      "no_neurons_per_layer", std::vector<unsigned int>(), "Number of neurons per layer.");
+  params.addParam<Real>("learning_rate", 0.001, "Learning rate (relaxation).");
+
+  return params;
+}
+
+BasicNNTrainer::BasicNNTrainer(const InputParameters & parameters)
```

Comment:
This seems like a lot of copy and paste from NearestPointTrainer. Should we just derive from that class and override `postTrain`?

Reply:
I simplified the NN trainer, the current version doesn't look like the NPTrainer.